src/distilabel/tasks/text_generation/evol_instruct.py

```diff
@@ -0,0 +1,155 @@
+
+    References:
+        - [`WizardLM: Empowering Large Language Models to Follow Complex Instructions`](https://arxiv.org/pdf/2304.12244.pdf)
+    """
+
+    system_prompt: str = ""
+
+    __jinja2_template__: str = _EVOL_INSTRUCT_TEMPLATE
+
+    def generate_prompt(
+        self, input: str, evolution_method: Optional[EvolutionMethod] = None, **_: Any
+    ) -> Prompt:
+        """Generates a prompt following the Evol-Instruct specification.
+
+        Args:
+            input (str): the input to be used for the prompt.
+            evolution_method (str, optional): The evolution method to be used. If not provided (the default), a random one is chosen
+                like the original paper. Available ones are "breadth", "constraints", "deepen", "concretizing" and "reasoning".
+
+        Returns:
+            Prompt: the generated prompt.
+
+        Examples:
+            >>> from distilabel.tasks.text_generation import EvolInstructTask
+            >>> task = EvolInstructTask()
+            >>> task.generate_prompt("Give three tips for staying healthy.")
+            Prompt(
+                system_prompt="",
+                formatted_prompt="I want you to act as a Prompt ...",
+            )
+        """
+        if not evolution_method:
+            evolution_method = random.choice(get_args(EvolutionMethod))
+
+        render_kwargs = {
+            "evol_method": evolution_method,
+            "instruction": input,
+        }
+        return Prompt(
+            system_prompt=self.system_prompt,
+            formatted_prompt=self.template.render(**render_kwargs),
+        )
+
+    @property
+    def output_args_names(self) -> List[str]:
+        return ["instruction"]
+
+    def _elimination_evolving(self, output: str) -> Optional[str]:
+        """Performs the elimination step of the Evol-Instruct task, steps 2-4 in the paper:
+
+        2. The evolved instruction makes it difficult for the LLM to generate a response. We found that
+        when the generated response contains “sorry” and is relatively short in length (i.e., less than
+        80 words), it often indicates that the LLM struggles to respond to the evolved instruction.
+        So we can use this rule to make a judgment.
+        3. The response generated by the LLM only contains punctuation and stop words.
+        4. The evolved instruction obviously copies some words from the evolving prompt, such as
+        “given prompt”, “rewritten prompt”, “#Rewritten Prompt#”, etc.
+        """
+        if output == "":
+            return
+
+        # 2) The evolved instruction makes it difficult for the LLM to generate a response.
+        if "sorry" in output.lower() and len(output.split(" ")) < 80:
+            return
+
+        # 3) The output only contains punctuation and stop words
+        if (
+            len(set(output).intersection(string.punctuation)) / len(set(output))
+        ) == 1.0:
+            logger.info(
```

Comment:
We might be able to use the CURL download approach to get a list of words or we should just skip it.?

Reply:
I forget about adding the stop words and only checked the punctuation signs. Originally was trying to set a threshold instead of checking for every character being a punctuation sign, but it implies dealing with that threshold... so I let it like that. I can try going with the curl download instead of adding a dependency for this

Reply:
Maybe I can just add this list of [stopwords](https://github.com/stopwords-iso/stopwords-en/blob/master/stopwords-en.txt) internally and avoid the download, let me know

Reply:
They don't really mention a threshold in the paper right?

Reply:
no, wil do just all the punctuation signs instead of the threshold

Reply:
Updated